### ISSUE TITLE
Increasing Upload Limits - Add to BOND?

### ISSUE THREAD

**Issue by petrowsky** [omega8cc/boa#752]
Well, I got tired of vim'ing into the conf files to bump upload limits - since I deal with video files over 100m.

I hacked up a quick shell script. Posting it here in case people find it or want to vote it up to see if it could be integrated into BOND. Don't know if I'm an edge case or this is common.

You'll obviously need to adjust the amount you want to bump by. I was too lazy to make a variable for that.
```
#!/bin/bash

# Change the php path based on the version you're using

_NGINX_CONF='/var/aegir/config/server_master/nginx.conf'
_PHP_CONF='/opt/php55/etc/php55.ini'

sed -i "s/\(.*client_max_body_size.*\)[0-9]\{3\}m;$/\1200m;/g" "${_NGINX_CONF}"
cat "${_NGINX_CONF}" | grep '_max_body_'
nginx -t

sed -i "s/.*post_max_size.*/post_max_size = 200M/g" "${_PHP_CONF}"
sed -i "s/.*upload_max_filesize.*/upload_max_filesize = 200M/g" "${_PHP_CONF}"
cat "${_PHP_CONF}" | grep '_max_'

echo "Run service nginx reload if no errors"
echo "Run service php55-fpm reload"
```

P.S. \1200 is not 1200. It's capture group \1 and 200m

**Comment by adamsro** [comment]
Not sure if this is the best way, but I added the following to /data/disk/o1/config/server_master/nginx/post.d/nginx_vhost_include.conf

    fastcgi_param PHP_VALUE "upload_max_filesize=3072M \n post_max_size=3072M"
    client_max_body_size          3072M;
    ...

The file is included by /data/disk/o1/config/includes/nginx_vhost_common.conf and not touched during upgrades.

I'm currently using the s3fs and s3fs_cors modules for the large file uploads, though, so I don't have to deal with large files on the servers disk or tie up a php worker.

**Comment by omega8cc** [comment]
I agree that the method suggested by @adamsro is currently better, because it allows to customize these limits just once and/or modify them without barracuda upgrade nor BOND nor extra scripts etc. We didn't make it easy to configure via standard variables because it is simply too dangerous since it opens the system to easy to launch DoS (POST) attacks.

**Comment by omega8cc** [comment]
We should add this to BOA docs, I think. Thanks @adamsro !

**Comment by olipud** [comment]
Wondering how to apply adamsro solution, each time i tried i get this:
nginx: [emerg] invalid number of arguments in "fastcgi_param" directive in /data/disk/o1/config/server_master/nginx/post.d/nginx_vhost_include.conf:4
I create nginx_vhost_include.conf, and put the code inside, restart nginx and get webserver off
Thank for your time

**Comment by olipud** [comment]
Just a forgotten ";" after post_max_size=3072M" !
fastcgi_param PHP_VALUE "upload_max_filesize=3072M \n post_max_size=3072M"
client_max_body_size          3072M;
